sumologic/resource_sumologic_cse_match_list.go

```diff
@@ -0,0 +1,284 @@
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_updated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_updated_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"items": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: false,
+						},
+						"expiration": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: false,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: false,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceSumologicCSEMatchListRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	var CSEMatchList *CSEMatchListGet
+	id := d.Id()
+
+	CSEMatchList, err := c.GetCSEMatchList(id)
+	if err != nil {
+		log.Printf("[WARN] CSE Match List not found when looking by id: %s, err: %v", id, err)
+
+	}
+
+	if CSEMatchList == nil {
+		log.Printf("[WARN] CSE Match List not found, removing from state: %v - %v", id, err)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", CSEMatchList.Name)
+	d.Set("default_ttl", CSEMatchList.DefaultTtl)
+	d.Set("description", CSEMatchList.Description)
+	d.Set("name", CSEMatchList.Name)
+	d.Set("target_column", CSEMatchList.TargetColumn)
+	d.Set("created", CSEMatchList.Created)
+	d.Set("created_by", CSEMatchList.CreatedBy)
+	d.Set("last_updated", CSEMatchList.LastUpdated)
+	d.Set("last_updated_by", CSEMatchList.LastUpdatedBy)
+
+	//items
+	var CSEMatchListItems *CSEMatchListItemsInMatchListGet
+
+	CSEMatchListItems, err2 := c.GetCSEMatchListItemsInMatchList(id)
+	if err2 != nil {
+		log.Printf("[WARN] CSE Match List items not found when looking by match list id: %s, err: %v", id, err2)
+	}
+	if CSEMatchListItems == nil {
+		d.Set("items", nil)
+	} else {
+		setItems(d, CSEMatchListItems.CSEMatchListItemsGetObjects)
+	}
+
+	return nil
+}
+
+func setItems(d *schema.ResourceData, items []CSEMatchListItemGet) {
+
+	var i []map[string]interface{}
+
+	for _, t := range items {
+		mapping := map[string]interface{}{
+			"id":          t.ID,
+			"description": t.Meta.Description,
+			"expiration":  t.Expiration,
+			"value":       t.Value,
+		}
+		i = append(i, mapping)
+	}
+
+	d.Set("items", i)
+
+}
+
+func resourceSumologicCSEMatchListDelete(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	//Match list items
+	itemsData := d.Get("items").([]interface{})
+	var items []CSEMatchListItemPost
+	for _, data := range itemsData {
+		items = append(items, resourceToCSEMatchListItem([]interface{}{data}))
+	}
+
+	if len(items) > 0 {
+		for _, item := range items {
+			err2 := c.DeleteCSEMatchListItem(item.ID)
```

Comment:
I don’t think you actually need to delete all of the items like this. If you delete the list, the items are deleted automatically in the background.

Reply:
thanks, done under https://github.com/SumoLogic/terraform-provider-sumologic/commit/dda976e0c3b3134f2ed3736bdaa8a69b129a6609